builtin/providers/aws/resource_aws_db_event_subscription_test.go

```diff
@@ -0,0 +1,100 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSDBEventSubscription_basic(t *testing.T) {
+	var v rds.DBInstance
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSDBEventSubscriptionDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSDBEventSubscriptionConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDBInstanceExists("aws_db_event_subscription.bar", &v),
+					testAccCheckAWSDBInstanceAttributes(&v),
+					resource.TestCheckResourceAttr(
+						"aws_db_instance.bar", "enabled", "true"),
+					resource.TestCheckResourceAttr(
+						"aws_db_instance.bar", "sns_topic", "arn:aws:sns:us-east-1:123456789012:RDS-Event"),
+					resource.TestCheckResourceAttr(
+						"aws_db_instance.bar", "source_type", "db-instance"),
+					resource.TestCheckResourceAttr(
+						"aws_db_instance.bar", "name", "baz"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSDBEventSubscriptionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).rdsconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_db_event_subscription" {
+			continue
+		}
+
+		var err error
+		resp, err := conn.DescribeEventSubscriptions(
+			&rds.DescribeEventSubscriptionsInput{
+				SubscriptionName: aws.String(rs.Primary.ID),
+			})
+
+		if ae, ok := err.(awserr.Error); ok && ae.Code() == "SubscriptionNotFound" {
+			continue
+		}
+
+		if err == nil {
+			if len(resp.EventSubscriptionsList) != 0 &&
+				*resp.EventSubscriptionsList[0].CustSubscriptionId == rs.Primary.ID {
+				return fmt.Errorf("Event Subscription still exists")
+			}
+		}
+
+		// Verify the error
+		newerr, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if newerr.Code() != "SubscriptionNotFound" {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var testAccAWSDBEventSubscriptionConfig = `
+resource "aws_db_event_subscription" "bar" {
+  name = "az"
```

Comment:
This should be more specific, ideally suggesting that this resource comes from terraform acceptance test - e.g. `name = "tf-acc-test-rds-event-subs"`.

It will be helpful when we end up looking for leaked resources if those tests happen to leak for any reason.